File: Stage04-Play/Lua/item.lua
local class = require "lib.class"
local Item = class:derive("Item") -- allows the use of print(object.type) -> "Item"

function Item:new(name, description , craftitems, uses, container)
	self.Name = name
	self.Description = description or ''
	self.Craftitems = craftitems or {}
	self.Uses = uses or 0
	self.Container = container or ''
end

function Item:GetName()
	return self.Name
end

function Item:GetDescription()
	return self.Description
end

function Item:SetDescription(value)
	self.Description = value
end

function Item:GetCraftitems()
	return self.Craftitems -- {'pliers', 'coat hanger'}
end

function Item:GetUses()
	return self.Uses
end

function Item:GetContainer()
	return self.Container
end

function Item:SetContainer(value)
	self.Container = value
end

return Item

File: Stage05-Enemies/C#/Item.cs
using System.Collections.Generic;

namespace Adventure_05_Weapon
{
    internal class Item
    {
        public string Name { get; set; } = "";
        public string Description { get; set; } = "";
        public List<string> CraftItems { get; set; } = new List<string>();
        public int Uses { get; set; } = 0;
        public string Container { get; set; } = "";
        public Item(string name, string description, List<string> craftitems, int uses, string container)
        {
            Name = name;
            Description = description; 
            CraftItems = craftitems;
            Uses = uses;
            Container = container;
        }
    }
}
